### PR TITLE
BLD,Cygwin: Include Python.h first in various C++ files

### PR DESCRIPTION
**Pull request by DWesl**
## PR summary

Python.h needs to be included before any system includes.  This is easiest if it is the first include.  

EDIT: `pybind11/pybind11.h` includes Python [three headers in](https://github.com/pybind/pybind11/issues/4999#issuecomment-1971076697).  There are two system headers included beforehand, if debugging with MSVC.  I didn't know that when I made this, and can remove `Python.h` from before `pybind11/pybind11.h` if desired.

Should fix the Cygwin problem noted in #27552

## PR checklist

- [N/A] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [X] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [N/A] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [N/A] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by DWesl** [comment]
Checked all files instead of going one-by-one using [check_python_h_first.py](https://github.com/matplotlib/matplotlib/files/14396346/check_python_h_first.txt) adapted from scipy/scipy#18049.

File: .github/workflows/cygwin.yml [hunk 1]
```diff
@@ -243,7 +243,7 @@ jobs:
         shell: bash.exe -eo pipefail -o igncr "{0}"
         id: cygwin-run-pytest
         run: |
-          xvfb-run pytest -rfEsXR -n auto \
+          xvfb-run python -m pytest -rfEsXR -n auto \
```
Comment (tacaswell) [inline .github/workflows/cygwin.yml:246]:
```suggestion
          xvfb-run pytest -rfEsXR -n auto \
```

See changes in https://github.com/matplotlib/matplotlib/pull/27552/files for details.  This is subtle interaction between pytest's import magic / test discovery + meson's build magic.

Comment (DWesl) [reply .github/workflows/cygwin.yml]: Well that explains the import errors once it found pytest.

**Review by ksunden — APPROVED** [review]

**Comment by QuLogic** [comment]
I think we should just consider `pybind11.h` to be like `Python.h` and ensure it is first (and so drop the extra `Python.h` additions if possible.)

File: src/_qhull_wrapper.cpp [hunk 2]
```diff
@@ -5,6 +5,7 @@
  * triangulation, construct an instance of the matplotlib.tri.Triangulation
  * class without specifying a triangles array.
  */
+#include <Python.h>
```
Comment (DWesl) [inline src/_qhull_wrapper.cpp:8]:
```suggestion
```

File: src/_path.h [hunk 3]
```diff
@@ -3,6 +3,7 @@
 #ifndef MPL_PATH_H
 #define MPL_PATH_H
 
+#include <Python.h>
```
Comment (DWesl) [inline src/_path.h:6]:
```suggestion
```

File: src/_image_wrapper.cpp [hunk 4]
```diff
@@ -1,3 +1,4 @@
+#include <Python.h>
```
Comment (DWesl) [inline src/_image_wrapper.cpp:1]:
```suggestion
```

File: src/_ttconv.cpp [hunk 5]
```diff
@@ -5,6 +5,7 @@
 
   Python wrapper for TrueType conversion library in ../ttconv.
  */
+#include <Python.h>
 #include "mplutils.h"
 
 #include <pybind11/pybind11.h>
```
Comment (DWesl) [inline src/_ttconv.cpp:11]:
```suggestion
#include <pybind11/pybind11.h>
#include "mplutils.h"
```

File: src/py_converters_11.h [hunk 6]
```diff
@@ -3,6 +3,7 @@
 
 // pybind11 equivalent of py_converters.h
 
+#include <Python.h>
```
Comment (DWesl) [inline src/py_converters_11.h:6]:
```suggestion
```

File: src/tri/_tri.h [hunk 7]
```diff
@@ -63,6 +63,7 @@
 #ifndef MPL_TRI_H
 #define MPL_TRI_H
 
+#include <Python.h>
```
Comment (DWesl) [inline src/tri/_tri.h:66]:
```suggestion
```

**Review by DWesl — COMMENTED** [review]
Remove `Python.h` includes immediately preceeding `pybind11/pybind11.h` includes.

**Review by tacaswell — APPROVED** [review]

**Pull request merged**